js/components/utils.js

```diff
@@ -431,34 +431,38 @@ var Utils = {
    * @function arrayDeepMerge
    * @param {Array} target - An array that will receive new items if additional items are passed
    * @param {Array} source - An array containing additional items to merge into target
-   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge function
+   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge, swap, unionBy, clone
    * @returns {Array} new merged array with items from both target and source
    */
   arrayDeepMerge: function(target, source, optionsArgument) {
+    var targetArray = optionsArgument.swap ? source : target;
```

Comment:
I tried that and it didn't maintain button sort order. we also need to swap target/source in the nested deep merge on line 456